yellowbrick/classifier.py

```diff
@@ -50,6 +51,149 @@ def __init__(self, model, ax=None, **kwargs):
 
         super(ClassificationScoreVisualizer, self).__init__(model, ax=ax, **kwargs)
 
+##########################################################################
+## ConfusionMatrix
+##########################################################################
+
+class ConfusionMatrix(ClassificationScoreVisualizer):
+    """
+    Creates a heatmap visualization of the sklearn.metrics.confusion_matrix().
+    Initialization: Requires a classification model
+    """
+    def __init__(self, model, ax=None, classes=None, **kwargs):
+        """
+        Provide a classifier model
+
+        Parameters
+        ----------
+        :param model: the Scikit-Learn estimator
+            Should be an instance of a classifier, else the __init__ will
+            return an error.
+
+        :param ax: the matplotlib axis to plot the figure on (if None, a new axis will be created)
+
+        :param classes: a list of class names to use in the confusion_matrix.
+            This is passed to the 'labels' parameter of sklearn.metrics.confusion_matrix(), and follows the behaviour
+            indicated by that function. It may be used to reorder or select a subset of labels.
+            If None, values that appear at least once in y_true or y_pred are used in sorted order.
+            Default: None
+        """
+        super(ConfusionMatrix, self).__init__(model, ax=ax, **kwargs)
+        #Parameters provided by super (for reference during development only):
+        #self.ax
+        #self.size
+        #self.color
+        #self.title
+        #self.estimator
+        #self.name
+
+        #Initialize all the other attributes we'll use (for coder clarity)
+        self.confusion_matrix = None
+
+        self.cmap = color_sequence(kwargs.pop('cmap', 'YlOrRd'))
+
+        #If possible, assign the classes. If this can't happen now, it will happen in the .fit() method
```

Comment:
Good point, I had forgotten about the need to not override user defined classes in fit(). We could simply use existing code and make sure to wrap .classes_ assignment in an if statement in the fit() method; but, making it a property instead is a good way to clue future developers in to the fact that they need to handle the assignment properly if classes is used anywhere else. I'll use the property approach. 

Another thing @rebeccabilbro and I discussed is a .util for checking if an estimator is already fit or not. It does not look like there is a tool in sklearn that does that already. Each estimator returns a NotFittedError if you call predict without fit, but each estimator type has a different way of deciding if it should return NotFittedError (e.g. KNeighbors checks for existence of self._fit_method, Forest checks for self.estimators_, linear classifiers check for self.coeff_).  Might just use .classes_ for the classifiers for now and make the .util when we need it. 

Is the _classes vs. classes_ the naming convention for properties vs. attributes, respectively?